syslog-ng-3.14.1 fails to build with automake-1.16.1 in a single job (make -j1)

# syslog-ng

## Version of syslog-ng

```
3.14.1
```

## Platform

Gentoo Linux

# Issue

## Failure

```
...
log/src/libevtlog_la-evtout.lo lib/eventlog/src/libevtlog_la-evtstr.lo lib/eventlog/src/libevtlog_la-evtctx.lo /libevtlog_la-evtstr.o lib/eventlog/src/.libs/libevtlog_la-evtctx.o lib/eventlog/src/.libs/libevtlog_la-evttags.o lib/eventlog/src/.libs/libevtlog_la-evtsyslog.o   -Wl,--as-needed  -mtune=native -O2 -pthread -Wl,-O1   -pthread -Wl,-soname -Wl,libevtlog-3.14.so.0 -o lib/eventlog/src/.libs/libevtlog-3.14.so.0.0.0
libtool: link: (cd "lib/eventlog/src/.libs" && rm -f "libevtlog-3.14.so.0" && ln -s "libevtlog-3.14.so.0.0.0" "libevtlog-3.14.so.0")
libtool: link: (cd "lib/eventlog/src/.libs" && rm -f "libevtlog.so" && ln -s "libevtlog-3.14.so.0.0.0" "libevtlog.so")
libtool: link: ( cd "lib/eventlog/src/.libs" && rm -f "libevtlog.la" && ln -s "../libevtlog.la" "libevtlog.la" )
/bin/sh ./libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc  -I/usr/include/uuid  -mtune=native -O2 -pipe -Wall -pthread -no-undefined -release 3.14 -version-info 0:0:0 -Wl,-O1 -Wl,--as-needed -o lib/libsyslog-ng.la -rpath /usr/lib64 lib/libsyslog_ng_la-afinter.lo lib/libsyslog_ng_la-alarms.lo lib/libsyslog_ng_la-apphook.lo lib/libsyslog_ng_la-block-ref-parser.lo lib/libsyslog_ng_la-cache.lo lib/libsyslog_ng_la-cfg.lo lib/libsyslog_ng_la-cfg-args.lo lib/libsyslog_ng_la-cfg-block.lo lib/libsyslog_ng_la-cfg-block-generator.lo lib/libsyslog_ng_la-cfg-lexer.lo lib/libsyslog_ng_la-cfg-lexer-subst.lo lib/libsyslog_ng_la-cfg-parser.lo lib/libsyslog_ng_la-cfg-tree.lo lib/libsyslog_ng_la-children.lo lib/libsyslog_ng_la-dnscache.lo lib/libsyslog_ng_la-driver.lo lib/libsyslog_ng_la-fdhelpers.lo lib/libsyslog_ng_la-file-perms.lo lib/libsyslog_ng_la-find-crlf.lo lib/libsyslog_ng_la-globals.lo lib/libsyslog_ng_la-gprocess.lo lib/libsyslog_ng_la-gsockaddr.lo lib/libsyslog_ng_la-gsocket.lo lib/libsyslog_ng_la-hostname.lo lib/libsyslog_ng_la-host-resolve.lo lib/libsyslog_ng_la-logmatcher.lo lib/libsyslog_ng_la-logmpx.lo lib/libsyslog_ng_la-logpipe.lo lib/libsyslog_ng_la-logqueue.lo lib/libsyslog_ng_la-logqueue-fifo.lo lib/libsyslog_ng_la-logreader.lo lib/libsyslog_ng_la-logsource.lo lib/libsyslog_ng_la-logstamp.lo lib/libsyslog_ng_la-logthrdestdrv.lo lib/libsyslog_ng_la-logwriter.lo lib/libsyslog_ng_la-mainloop.lo lib/libsyslog_ng_la-mainloop-call.lo lib/libsyslog_ng_la-mainloop-worker.lo lib/libsyslog_ng_la-mainloop-io-worker.lo lib/libsyslog_ng_la-module-config.lo lib/libsyslog_ng_la-memtrace.lo lib/libsyslog_ng_la-messages.lo lib/libsyslog_ng_la-ml-batched-timer.lo lib/libsyslog_ng_la-msg-format.lo lib/libsyslog_ng_la-parse-number.lo lib/libsyslog_ng_la-pathutils.lo lib/libsyslog_ng_la-persist-state.lo lib/libsyslog_ng_la-plugin.lo lib/libsyslog_ng_la-poll-events.lo lib/libsyslog_ng_la-poll-fd-events.lo lib/libsyslog_ng_la-pragma-parser.lo lib/libsyslog_ng_la-persistable-state-presenter.lo lib/libsyslog_ng_la-rcptid.lo lib/libsyslog_ng_la-reloc.lo lib/libsyslog_ng_la-run-id.lo lib/libsyslog_ng_la-scratch-buffers.lo lib/libsyslog_ng_la-serialize.lo lib/libsyslog_ng_la-service-management.lo lib/libsyslog_ng_la-str-format.lo lib/libsyslog_ng_la-str-utils.lo lib/libsyslog_ng_la-syslog-names.lo lib/libsyslog_ng_la-string-list.lo lib/libsyslog_ng_la-timeutils.lo lib/libsyslog_ng_la-type-hinting.lo lib/libsyslog_ng_la-ringbuffer.lo lib/libsyslog_ng_la-late_ack_tracker.lo lib/libsyslog_ng_la-early_ack_tracker.lo lib/libsyslog_ng_la-crypto.lo lib/libsyslog_ng_la-tlscontext.lo lib/libsyslog_ng_la-uuid.lo lib/libsyslog_ng_la-userdb.lo lib/libsyslog_ng_la-utf8utils.lo lib/transport/libsyslog_ng_la-transport-tls.lo lib/libsyslog_ng_la-host-id.lo lib/libsyslog_ng_la-resolved-configurable-paths.lo lib/libsyslog_ng_la-cfg-lex.lo lib/libsyslog_ng_la-cfg-grammar.lo lib/libsyslog_ng_la-block-ref-grammar.lo lib/libsyslog_ng_la-pragma-grammar.lo lib/scanner/csv-scanner/libsyslog_ng_la-csv-scanner.lo lib/scanner/kv-scanner/libsyslog_ng_la-kv-scanner.lo lib/scanner/list-scanner/libsyslog_ng_la-list-scanner.lo lib/transport/libsyslog_ng_la-logtransport.lo lib/transport/libsyslog_ng_la-transport-aux-data.lo lib/transport/libsyslog_ng_la-transport-file.lo lib/transport/libsyslog_ng_la-transport-pipe.lo lib/transport/libsyslog_ng_la-transport-socket.lo lib/logproto/libsyslog_ng_la-logproto-client.lo lib/logproto/libsyslog_ng_la-logproto-server.lo lib/logproto/libsyslog_ng_la-logproto-buffered-server.lo lib/logproto/libsyslog_ng_la-logproto-dgram-server.lo lib/logproto/libsyslog_ng_la-logproto-framed-client.lo lib/logproto/libsyslog_ng_la-logproto-framed-server.lo lib/logproto/libsyslog_ng_la-logproto-text-client.lo lib/logproto/libsyslog_ng_la-logproto-text-server.lo lib/logproto/libsyslog_ng_la-logproto-indented-multiline-server.lo lib/logproto/libsyslog_ng_la-logproto-regexp-multiline-server.lo lib/logproto/libsyslog_ng_la-logproto-multiline-server.lo lib/logproto/libsyslog_ng_la-logproto-record-server.lo lib/logproto/libsyslog_ng_la-logproto-builtins.lo lib/filter/libsyslog_ng_la-filter-expr.lo lib/filter/libsyslog_ng_la-filter-op.lo lib/filter/libsyslog_ng_la-filter-cmp.lo lib/filter/libsyslog_ng_la-filter-in-list.lo lib/filter/libsyslog_ng_la-filter-tags.lo lib/filter/libsyslog_ng_la-filter-netmask.lo lib/filter/libsyslog_ng_la-filter-netmask6.lo lib/filter/libsyslog_ng_la-filter-call.lo lib/filter/libsyslog_ng_la-filter-re.lo lib/filter/libsyslog_ng_la-filter-pri.lo lib/filter/libsyslog_ng_la-filter-pipe.lo lib/filter/libsyslog_ng_la-filter-expr-parser.lo lib/filter/libsyslog_ng_la-filter-expr-grammar.lo lib/parser/libsyslog_ng_la-parser-expr.lo lib/parser/libsyslog_ng_la-parser-expr-parser.lo lib/parser/libsyslog_ng_la-parser-expr-grammar.lo lib/rewrite/libsyslog_ng_la-rewrite-expr.lo lib/rewrite/libsyslog_ng_la-rewrite-set-tag.lo lib/rewrite/libsyslog_ng_la-rewrite-set.lo lib/rewrite/libsyslog_ng_la-rewrite-unset.lo lib/rewrite/libsyslog_ng_la-rewrite-subst.lo lib/rewrite/libsyslog_ng_la-rewrite-expr-parser.lo lib/rewrite/libsyslog_ng_la-rewrite-expr-grammar.lo lib/rewrite/libsyslog_ng_la-rewrite-groupset.lo lib/template/libsyslog_ng_la-templates.lo lib/template/libsyslog_ng_la-macros.lo lib/template/libsyslog_ng_la-simple-function.lo lib/template/libsyslog_ng_la-repr.lo lib/template/libsyslog_ng_la-compiler.lo lib/template/libsyslog_ng_la-user-function.lo lib/template/libsyslog_ng_la-escaping.lo lib/value-pairs/libsyslog_ng_la-value-pairs.lo lib/value-pairs/libsyslog_ng_la-transforms.lo lib/value-pairs/libsyslog_ng_la-cmdline.lo lib/value-pairs/libsyslog_ng_la-evttag.lo lib/stats/libsyslog_ng_la-stats.lo lib/stats/libsyslog_ng_la-stats-control.lo lib/stats/libsyslog_ng_la-stats-counter.lo lib/stats/libsyslog_ng_la-stats-cluster.lo lib/stats/libsyslog_ng_la-stats-csv.lo lib/stats/libsyslog_ng_la-stats-log.lo lib/stats/libsyslog_ng_la-stats-registry.lo lib/stats/libsyslog_ng_la-stats-syslog.lo lib/stats/libsyslog_ng_la-stats-query.lo lib/stats/libsyslog_ng_la-stats-query-commands.lo lib/stats/libsyslog_ng_la-stats-cluster-logpipe.lo lib/stats/libsyslog_ng_la-stats-cluster-single.lo lib/control/libsyslog_ng_la-control-commands.lo lib/control/libsyslog_ng_la-control-main.lo lib/control/libsyslog_ng_la-control-server.lo lib/debugger/libsyslog_ng_la-debugger.lo lib/debugger/libsyslog_ng_la-tracer.lo lib/debugger/libsyslog_ng_la-debugger-main.lo lib/compat/libsyslog_ng_la-getutent.lo lib/compat/libsyslog_ng_la-inet_aton.lo lib/compat/libsyslog_ng_la-memrchr.lo lib/compat/libsyslog_ng_la-pio.lo lib/compat/libsyslog_ng_la-glib.lo lib/compat/libsyslog_ng_la-strcasestr.lo lib/compat/libsyslog_ng_la-strtok_r.lo lib/compat/libsyslog_ng_la-time.lo lib/compat/libsyslog_ng_la-openssl_support.lo lib/compat/libsyslog_ng_la-getent.lo lib/logmsg/libsyslog_ng_la-gsockaddr-serialize.lo lib/logmsg/libsyslog_ng_la-logmsg.lo lib/logmsg/libsyslog_ng_la-logmsg-serialize.lo lib/logmsg/libsyslog_ng_la-logmsg-serialize-fixup.lo lib/logmsg/libsyslog_ng_la-nvhandle-descriptors.lo lib/logmsg/libsyslog_ng_la-nvtable.lo lib/logmsg/libsyslog_ng_la-nvtable-serialize.lo lib/logmsg/libsyslog_ng_la-nvtable-serialize-legacy.lo lib/logmsg/libsyslog_ng_la-tags-serialize.lo lib/logmsg/libsyslog_ng_la-timestamp-serialize.lo lib/logmsg/libsyslog_ng_la-tags.lo lib/str-repr/libsyslog_ng_la-decode.lo lib/str-repr/libsyslog_ng_la-encode.lo -lrt -Wl,--export-dynamic -lgmodule-2.0 -pthread -lgthread-2.0 -pthread -lglib-2.0 -Wl,--whole-archive -L./lib/eventlog/src -levtlog -Wl,--no-whole-archive   -lpcre  -ldl -livykis  ./lib/secret-storage/libsecret-storage.la -lssl -lcrypto -luuid
libtool:   error: cannot find the library './lib/secret-storage/libsecret-storage.la' or unhandled argument './lib/secret-storage/libsecret-storage.la'
make[2]: *** [Makefile:8563: lib/libsyslog-ng.la] Error 1
make[1]: *** [Makefile:17572: all-recursive] Error 1
make: *** [Makefile:7588: all] Error 2
```

## Steps to reproduce

```
1. install automake 1.16.1
2. build syslog-ng
2.1. rm -rv lib/ivykis modules/afmongodb/mongo-c-driver modules/afamqp/rabbitmq-c
2.2. autoreconf --install
2.3. ./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --disable-dependency-tracking --disable-silent-rules --docdir=/usr/share/doc/syslog-ng-3.14.1 --htmldir=/usr/share/doc/syslog-ng-3.14.1/html --libdir=/usr/lib64 --disable-docs --disable-java --disable-java-modules --disable-riemann --enable-manpages --localstatedir=/var/lib/syslog-ng --sysconfdir=/etc/syslog-ng --with-embedded-crypto --with-ivykis=system --with-module-dir=/usr/lib64/syslog-ng --with-pidfile-dir=/var/run --with-systemdsystemunitdir=/lib/systemd/system --disable-amqp --without-librabbitmq-client --disable-linux-caps --disable-sql --disable-geoip --disable-http --enable-ipv6 --disable-json --disable-mongodb --without-mongoc --disable-legacy-mongodb-options --disable-pacct --disable-python --disable-redis --disable-smtp --disable-spoof-source --disable-systemd --enable-tcp-wrapper
2.4. make -j1
```

It works when we use multiple jobs (make -j4 for example). It also works with automake 1.15.

Gentoo bug report: https://bugs.gentoo.org/650172

--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -47,7 +47,7 @@ lib_libsyslog_ng_la_LDFLAGS		= -no-undefined -release ${LSNG_RELEASE} \
 
 lib_test_subdirs			= lib_filter lib_logproto lib_parser lib_rewrite lib_template lib_stats lib_control
 
-lib_libsyslog_ng_la_DEPENDENCIES	= lib/eventlog/src/libevtlog.la
+lib_libsyslog_ng_la_DEPENDENCIES       = lib/eventlog/src/libevtlog.la lib/secret-storage/libsecret-storage.la
 
 if IVYKIS_INTERNAL
 lib_libsyslog_ng_la_DEPENDENCIES	+= lib/ivykis/src/libivykis.la

